core/src/main/java/io/questdb/std/json/JsonParser.java

```diff
@@ -0,0 +1,221 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2024 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.std.json;
+
+import io.questdb.std.Os;
+import io.questdb.std.QuietCloseable;
+import io.questdb.std.bytes.NativeByteSink;
+import io.questdb.std.str.DirectUtf8Sequence;
+import io.questdb.std.str.DirectUtf8Sink;
+
+public class JsonParser implements QuietCloseable {
+    public static final int SIMDJSON_PADDING;
+    private long impl;
+
+    public JsonParser() {
+        impl = create();
+    }
+
+    /**
+     * Convert a JSON path to the SIMDJSON pointer format.
+     */
+    public static void convertJsonPathToPointer(DirectUtf8Sequence path, DirectUtf8Sink dest) {
+        try (NativeByteSink destSink = dest.borrowDirectByteSink()) {
+            if (path.size() > 0) {
+                convertJsonPathToPointer(path.ptr(), path.size(), destSink.ptr());
+            }
+        }
+    }
+
+    @Override
+    public void close() {
+        if (impl != 0) {
+            destroy(impl);
+            impl = 0;
+        }
+    }
+
+    /**
+     * Get a pointer value and force the result to a string, regardless of the type.
+     */
+    public void queryPointer(DirectUtf8Sequence json, DirectUtf8Sequence pointer, JsonResult result, DirectUtf8Sink dest, int maxSize, long defaultValuePtr, long defaultValueSize) {
+        assert json.tailPadding() >= SIMDJSON_PADDING;
+        final long nativeByteSinkPtr = dest.borrowDirectByteSink().ptr();
+        if (!(dest.capacity() - dest.size() >= maxSize)) {
+            throw new IllegalArgumentException("Destination buffer is too small");
```

Comment:
this must be user readable CairoException. We do not throw illegal argument exception at the user